llvm/lib/Target/SPIRV/SPIRVModuleAnalysis.cpp

```diff
@@ -884,6 +902,12 @@ void addInstrRequirements(const MachineInstr &MI,
   default:
     break;
   }
+
+  // If we require capability Shader, then we can remove the requirement for
+  // the BitInstructions capability, since Shader is a superset capability
+  // of BitInstructions.
+  Reqs.removeCapabilityIf(SPIRV::Capability::Shader,
+                          SPIRV::Capability::BitInstructions);
```

Comment:
What if we can use `SPV_KHR_bit_instructions`, but `Shader` was added because of the presence of `OpTypeRuntimeArray`? Should `Shader` be removed in this case?

Reply:
yes, that's my reading. Shader is a superset so afaiu there's no point in having both. If something is required in Shader that's not in bit_instructions, then lets just drop bit_instructions and use Shader instead.

Reply:
Well, but your implementation removes `Shader` in the presence of `BitInstructions` which is not expected according to your description, right?

Also it would be nice to have a test on this feature (not required in this patch).

By the way SPIRV translator leaves both capabilities. I added your test to another existing test in translator's test base and ran it with -spirv-ext=+SPV_KHR_bit_instructions. Here is the result:
```
...
               OpCapability Matrix
               OpCapability Shader
               OpCapability Addresses
               OpCapability Linkage
               OpCapability Kernel
               OpCapability BitInstructions
               OpExtension "SPV_KHR_bit_instructions"
...
```
However I agree that it's excessive. I haven't updated the translator for a long time, so maybe this behavior has already changed.

Reply:
Thanks for catching that. I passed the arguments the wrong way around.

Reply:
Please also clang-format the last changes.